src/bridges/erc4626/VaultBridge.sol

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: GPL-2.0-only
+// Copyright 2020 Spilsbury Holdings Ltd
+pragma solidity >=0.6.10 <=0.8.10;
+pragma experimental ABIEncoderV2;
+
+import {SafeMath} from '@openzeppelin/contracts/utils/math/SafeMath.sol';
+
+import {IDefiBridge} from '../../interfaces/IDefiBridge.sol';
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/access/Ownable.sol';
+import {AztecTypes} from '../../aztec/AztecTypes.sol';
+import {IERC4626} from './Interfaces/IERC4626.sol';
+
+/**
+ * @title Aztec Connect Bridge for ERC4626 compatible vaults
+ * @author (@johhonn on Github)
+ * @notice You can use this contract to stake asset tokens and bridge share tokens for any ERC2626 compatible bride.
+ * @dev Implementation of the IDefiBridge interface for IERC4626
+ *
+ 
+ */
+contract VaultBridge is IDefiBridge, Ownable {
+    using SafeMath for uint256;
+
+    address public immutable rollupProcessor;
+
+    constructor(address _rollupProcessor) public {
+        rollupProcessor = _rollupProcessor;
+    }
+
+    /**
+     * @notice Function which stakes or unstakes
+     * @dev This method can only be called from the RollupProcessor.sol. If the input asset is the ERC4626 vault asset, staking flow is
+     * executed. If input is the vault share token, unstaking. RollupProcessor.sol has to transfer the tokens to the bridge before calling
+     * the method. If this is not the case, the function will revert (either in STAKING_CONTRACT.stake(...) or during
+     * SB burn).
+     *
+     * @param inputAssetA - Vault asset (Staking) or Vault Address (Unstaking)
+     * @param outputAssetA - Vault Address (Staking) or Vault asset (UnStaking)
+     * @param totalInputValue - the amount of assets to stake or shares to unstake
+     * @return outputValueA - the amount of shares (Staking) or assets (Unstaking) minted/transferred to
+     * the RollupProcessor.sol
+     */
+    function convert(
+        AztecTypes.AztecAsset memory inputAssetA,
+        AztecTypes.AztecAsset memory inputAssetB,
+        AztecTypes.AztecAsset memory outputAssetA,
+        AztecTypes.AztecAsset memory outputAssetB,
+        uint256 totalInputValue,
+        uint256 interactionNonce,
+        uint64 auxData,
+        address rollupBeneficiary
+    )
+        external
+        payable
+        override
+        returns (
+            uint256 outputValueA,
+            uint256 outputValueB,
+            bool isAsync
+        )
+    {
+        // // ### INITIALIZATION AND SANITY CHECKS
+        require(msg.sender == rollupProcessor, 'ExampleBridge: INVALID_CALLER');
```

Comment:
Since solidity 0.8.4 there is a support for custom errors. It would be nice if you used them instead of these require statements. Their advantage is that it decreases deployment costs because it doesn't store the error string in the bytecode but instead it only stores a 4 byte identifier pointing to the error definition in the ABI.